### PR TITLE
refactor: Remove GravsearchQueryOptimisationFeature and simplify

### PR DESCRIPTION
**Pull request by seakayone**
### Task Description/Number

Issue Number: DEV-

### PR Type

- [ ] fix: represents bug fixes
- [x] refactor: represents production code refactoring
- [ ] feat: represents a new feature
- [ ] docs: documentation changes (no production code change)
- [ ] chore: maintenance tasks (no production code change)
- [ ] test: all about tests: adding, refactoring tests (no production code change)
- [ ] other... Please describe:

### Basic Requirements for bug fixes and features

- [ ] Tests for the changes have been added
- [ ] Docs have been added / updated

### Does this PR introduce a breaking change?

- [ ] Yes

### Does this PR change client-test-data?

- [ ] Yes

### CODE REVIEW

**Review by BalduinLandolt — APPROVED** [review]
I don't understand 100% of what I read, but seems sensible and looks much cleaner

File: webapi/src/main/scala/org/knora/webapi/messages/util/search/SparqlQuery.scala [hunk 1]
```diff
@@ -91,7 +91,7 @@ case class Count(inputVariable: QueryVariable, distinct: Boolean, outputVariable
 
   val outputVariable: QueryVariable = QueryVariable(outputVariableName)
 
-  val distinctAsStr: String = if (distinct) {
+  private val distinctAsStr: String = if (distinct) {
     "DISTINCT"
   } else {
     ""
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/messages/util/search/SparqlQuery.scala:97]:
```suggestion
  private val distinctAsStr: String = if (distinct) "DISTINCT" else ""
```

File: webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/prequery/GravsearchQueryOptimisation.scala [hunk 2]
```diff
@@ -141,18 +102,18 @@ class RemoveRedundantKnoraApiResourceOptimisationFeature(
...
   /**
    * Performs the optimisation.
    *
    * @param patterns the query patterns.
    * @return the optimised query patterns.
    */
-  override def optimiseQueryPatterns(patterns: Seq[QueryPattern]): Seq[QueryPattern] = {
+  def removeEntitiesInferredFromProperty(
+    patterns: Seq[QueryPattern],
+    typeInspectionResult: GravsearchTypeInspectionResult
```
Comment (mpro7) [inline webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/prequery/GravsearchQueryOptimisation.scala:115]: Either remove params from Scaladocs or update them.

**Review by mpro7 — APPROVED** [review]
There are two comments from my site:

File: webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/prequery/GravsearchQueryOptimisation.scala (hunk 2 above)
Comment (seakayone) [reply webapi/src/main/scala/org/knora/webapi/messages/util/search/gravsearch/prequery/GravsearchQueryOptimisation.scala:115]: Will do in a future PR: 9f04f05c6320d12c84a840fe5d99a49f29f99703

**Pull request merged**